app/components/inventory-number-input.js

```diff
@@ -4,27 +4,58 @@ import AjaxPromise from '../utils/ajax-promise';
 const { getOwner } = Ember;
 
 export default Ember.Component.extend({
-
+  i18n: Ember.inject.service(),
   attributeBindings: ["name", "inputId", "value", "invalid", "disabled", "packageId"],
   isCordovaApp: config.cordova.enabled,
   messageBox: Ember.inject.service(),
   showMenu: false,
   bardcodeReadonly: true,
 
+  checkPermissionAndScan() {
+    let _this = this;
+    let permissions = cordova.plugins.permissions;
+    let permissionError = () => {
+      let error_message = this.get("i18n").t("camera_scan.permission_error");
+      _this.get("messageBox").alert(error_message);
```

Comment:
Is `this` not available here ? you are using fat arrow function it should be  available inside it.